src/devices_models/devices/common/cost_functions.jl

```diff
@@ -376,7 +376,11 @@ function ps_cost(
     gen_cost = JuMP.GenericAffExpr{Float64, _variable_type(psi_container)}()
     for (t, var) in enumerate(variable)
         if !isnothing(bin)
-            c, pwl_vars = _pwl_cost(psi_container, var, cost_array, bin[t])
+            if bin isa ParameterJuMP.ParameterRef
```

Comment:
Please add detailed comments in the code about the logic here, I think we will have to do some major refactoring here and knowing the original intention will be a good thing to have.